google/resource_kms_key_ring_test.go

```diff
@@ -0,0 +1,78 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"log"
+)
+
+func TestAccGoogleKmsKeyRing_basic(t *testing.T) {
+	name := acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testGoogleKmsKeyRing_recreate(name),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testGoogleKmsKeyRing_basic(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleKmsKeyRingExists("google_kms_key_ring.key_ring"),
+				),
+			},
+		},
+	})
+}
+
+func testGoogleKmsKeyRing_basic(name string) string {
+	return fmt.Sprintf(`
+	resource "google_kms_key_ring" "key_ring" {
+		name = "%s"
+		location = "us-central1"
+	}
+	`, name)
+}
+
+func testAccCheckGoogleKmsKeyRingExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Resource not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		location := rs.Primary.Attributes["location"]
+
+		parent := createKmsResourceParentString(config.Project, location)
+		keyRingName := createKmsResourceKeyRingName(config.Project, location, name)
+
+		listKeyRingsResponse, err := config.clientKms.Projects.Locations.KeyRings.List(parent).Do()
```

Comment:
I chose to use `List` and iterate over the results instead of using `Get` on the given KeyRing id, because otherwise it would have looked too much like the production code in `resourceKmsKeyRingRead`